spec/features/threading_spec.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+fixture_path = 'spec/fixtures'
+
+describe 'thread safety checks' do
+  let(:correct_sizes) {
+    {
+      'basic.csv' => 5,
+      'simple_with_header.csv' => 4,
+      'emoji.csv' => 3,
+      'quoted.csv' => 4
+    }
+  }
+
+  it 'at least returns the right number of results from each thread' do
```

Comment:
If you run this same spec on master, you'll see something like this (it's possible it could produce non-deterministic results)

```
Failures:

  1) thread safety checks at least returns the right number of results from each thread
     Failure/Error: expect(d[1].size).to eq(correct_sizes[d[0]])

       expected: 5
            got: 16

       (compared using ==)
     # ./spec/features/threading_spec.rb:22:in `block (3 levels) in <top (required)>'
     # ./spec/features/threading_spec.rb:21:in `each'
     # ./spec/features/threading_spec.rb:21:in `block (2 levels) in <top (required)>'
```

Reply:
@jpcamara thank you for finding this issue, and your PR!